src/components/flex/flex_item.tsx

```diff
@@ -61,18 +96,38 @@ export const OuiFlexItem: FunctionComponent<
 > = ({
   children,
   className,
-  grow = true,
+  grow = true, // true -> use default behaviour 'flex-grow: 1'
+  shrink = true, //  true -> use default behaviour 'flex-shrink: 1'
+  basis = true, //  true -> use default behaviour 'flex-basis: 0%'
   component: Component = 'div',
   ...rest
 }) => {
   validateGrowValue(grow);
+  validateShrinkValue(shrink);
+  validateBasisValue(basis);
 
   const classes = classNames(
     'ouiFlexItem',
     {
+      // use flex-grow: 0 and flex-basis: auto
       'ouiFlexItem--flexGrowZero': !grow,
+      // use flex-shrink: 0 and flex-basis: auto
+      'ouiFlexItem--flexShrinkZero': !shrink,
+      // use flex-basis: auto
+      'ouiFlexItem--flexBasisAuto': !basis,
```

Comment:
I think this should check `basis === false` to remove default class being added to preserve existing functionality with `grow` semantics

Reply:
But the expression `!basis` is almost the same as `basis === false `. Both when `basis` is `false` evaluates to `true` (`!false` is `true` and `false === false` is `true`). The only minor difference is when `basis`  is  `undefined`. In that case, the first expression (`!basis`) will be `true`, meanwhile the second one (`basis === false`), will be `false`. I was just following the same approach for the classes `ouiFlexItem--flexGrowZero` and `ouiFlexItem--flexShrinkZero`.

Reply:
Exactly, branching on my comment above, this class should only be applied if the builder explicitly specifies a `basis` value

Reply:
Code updated!